sympy/matrices/expressions/matpow.py

```diff
@@ -4,18 +4,19 @@
 from sympy.core.sympify import _sympify
 from sympy.core.compatibility import range
 from sympy.matrices import MatrixBase
-from sympy.core import S
+from sympy.core import S, Symbol
 
 
 class MatPow(MatrixExpr):
 
-    def __new__(cls, base, exp):
+    def __new__(cls, base, exp, evaluate=False):
```

Comment:
MatPow is an exception.The normal behaviour of MatPow is to be unevaluated(that is actually the purpose of it). So , maybe it is best to create that global parameter for matrix powers.

Currently we have : 
```
>>X = Matrix([[1, 2],[3, 4]])
>>> with evaluate(True):
...     X**2
... 
Matrix([
[ 7, 10],
[15, 22]])
>>> with evaluate(False):
...     X**2
... 
Matrix([
[0*(3 + 3*4) + 1 + 2*3, 0*(2*3 + 4*4) + 2 + 2*4],
[0*(1 + 2*3) + 3 + 3*4, 0*(2 + 2*4) + 2*3 + 4*4]])
```
The evaluate check percolates down to the elements. Would it be desirable behaviour to have it return `X**2` in the `evaluate(False)` block ?